src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/KeyedDependencyInjectionSpecificationTests.cs

```diff
@@ -324,6 +324,41 @@ public void ResolveKeyedServiceTransientTypeWithAnyKey()
             Assert.NotSame(first, second);
         }
 
+        [Fact]
+        public void ResolveKeyedServiceFromInjectedServiceProvider()
```

Comment:
Am I correct in assuming that this test would have passed before these ServiceProviderEngineScope changes?

Reply:
Nope, that was failing, see https://github.com/dotnet/runtime/issues/89447#issuecomment-1650548216